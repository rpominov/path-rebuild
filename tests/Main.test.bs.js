// Generated by ReScript, PLEASE EDIT WITH CARE
'use strict';

var Jest = require("./Jest.bs.js");
var Curry = require("@rescript/std/lib/js/curry.js");
var Js_exn = require("@rescript/std/lib/js/js_exn.js");
var Belt_Result = require("@rescript/std/lib/js/belt_Result.js");
var PathRebuild = require("../PathRebuild.bs.js");

function msg(err) {
  if (err.TAG === /* Ok */0) {
    return Js_exn.raiseError("Not a Error(_)");
  } else {
    return err._0;
  }
}

Jest.each([
      "test/{1..0}/{-2}.js",
      "test/{2..1}/{-2}.js",
      "test/{-1..1}/{-2}.js",
      "test/{-1..-2}/{-2}.js",
      "test/{1..0a}/{-2}.js",
      "test/{0a}/{-2}.js",
      "{-2}.js\\",
      "foo/{-2..",
      "test/{1..2.1}/{-2}.js",
      "test/{1...2}/{-2}.js",
      "test/{1.2}/{-2}.js",
      "test/{1\\..2}/{-2}.js",
      "test/{1{..2}/{-2}.js",
      "test/{1/..2}/{-2}.js",
      "test}/{1..2}/{-2}.js"
    ], "Parse errors", (function (pattern) {
        expect(msg(PathRebuild.make(pattern))).toMatchSnapshot();
        
      }));

Jest.each2([
      [
        "{0..-3}/{-2}.js",
        "file.sql"
      ],
      [
        "{0..-4}/{-2}.js",
        "a/b/c/d/file.sql"
      ]
    ], "Transform", (function (pattern, path) {
        var transform = Belt_Result.getExn(PathRebuild.make(pattern));
        expect(Belt_Result.getExn(Curry._2(transform, "/", path))).toMatchSnapshot();
        
      }));

test("Unconventional separator", (function () {
        var transform = Belt_Result.getExn(PathRebuild.make("{0..-4}/{-2}.js"));
        expect(Belt_Result.getExn(Curry._2(transform, "#", "a#b#c#d#file.sql"))).toMatchSnapshot();
        
      }));

test("Default separator", (function () {
        var transform = Belt_Result.getExn(PathRebuild.make("{0..-4}/{-2}.js"));
        expect(Belt_Result.getExn(Curry._2(transform, undefined, "a/b/c/d/file.sql"))).toEqual("a/b/c/file.js");
        
      }));

test("Absolute path", (function () {
        var transform = Belt_Result.getExn(PathRebuild.make("{0..-3}/{-2}.js"));
        expect(msg(Curry._2(transform, undefined, "/file.sql"))).toMatchSnapshot();
        
      }));

exports.msg = msg;
/*  Not a pure module */
