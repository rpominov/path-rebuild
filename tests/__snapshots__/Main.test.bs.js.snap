// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Absolute path 1`] = `"An absolute path cannot be used as a source path"`;

exports[`Parse errors (0) "test/{1..0}/{-2}.js" 1`] = `
"test/{1..0}/{-2}.js
          ^
Bad range limits: 1..0"
`;

exports[`Parse errors (1) "test/{2..1}/{-2}.js" 1`] = `
"test/{2..1}/{-2}.js
          ^
Bad range limits: 2..1"
`;

exports[`Parse errors (2) "test/{-1..1}/{-2}.js" 1`] = `
"test/{-1..1}/{-2}.js
           ^
Bad range limits: -1..1"
`;

exports[`Parse errors (3) "test/{-1..-2}/{-2}.js" 1`] = `
"test/{-1..-2}/{-2}.js
            ^
Bad range limits: -1..-2"
`;

exports[`Parse errors (4) "test/{1..0a}/{-2}.js" 1`] = `
"test/{1..0a}/{-2}.js
           ^
Bad range limit: 0a"
`;

exports[`Parse errors (5) "test/{0a}/{-2}.js" 1`] = `
"test/{0a}/{-2}.js
        ^
Bad range limit: 0a"
`;

exports[`Parse errors (6) "{-2}.js\\\\" 1`] = `
"{-2}.js\\\\
        ^
Unexpected end of string. Expected a character after the escape symbol"
`;

exports[`Parse errors (7) "foo/{-2.." 1`] = `
"foo/{-2..
         ^
Unexpected end of string. Did you forget to close a range?"
`;

exports[`Parse errors (8) "test/{1..2.1}/{-2}.js" 1`] = `
"test/{1..2.1}/{-2}.js
          ^
Unexpected . symbol"
`;

exports[`Parse errors (9) "test/{1...2}/{-2}.js" 1`] = `
"test/{1...2}/{-2}.js
         ^
Unexpected . symbol"
`;

exports[`Parse errors (10) "test/{1.2}/{-2}.js" 1`] = `
"test/{1.2}/{-2}.js
        ^
Unexpected character: 2. Was expecting a . symbol"
`;

exports[`Parse errors (11) "test/{1\\\\..2}/{-2}.js" 1`] = `
"test/{1\\\\..2}/{-2}.js
       ^
Unexpected escape symbol inside a range"
`;

exports[`Parse errors (12) "test/{1{..2}/{-2}.js" 1`] = `
"test/{1{..2}/{-2}.js
       ^
Unexpected { symbol inside a range"
`;

exports[`Parse errors (13) "test/{1/..2}/{-2}.js" 1`] = `
"test/{1/..2}/{-2}.js
       ^
Unexpected / symbol inside a range"
`;

exports[`Parse errors (14) "test}/{1..2}/{-2}.js" 1`] = `
"test}/{1..2}/{-2}.js
    ^
Unexpected } symbol"
`;

exports[`Transform (0) "{0..-3}/{-2}.js" "file.sql" 1`] = `"file.js"`;

exports[`Transform (1) "{0..-4}/{-2}.js" "a/b/c/d/file.sql" 1`] = `"a/b/c/file.js"`;

exports[`Unconventional separator 1`] = `"a#b#c#file.js"`;
